### PR TITLE
Scala Stream Collector: Fix incompatible jackson dependencies

### PR DESCRIPTION
**Pull request by oguzhanunlu**

### CODE REVIEW

**Comment by oguzhanunlu** [comment]
Hi @benjben , I used [`sbt-dependency-graph`](https://github.com/jrudolph/sbt-dependency-graph) plugin to check dependency versions and found the version mismatch between `jackson-core` and `jackson-dataformat-cbor`. The one with the lower version was `jackson-dataformat-cbor` and I added its' `2.9.10` version to align with jackson-core.

You're right, I'll add a note now.

**Comment by oguzhanunlu** [comment]
I added the note and force pushed @benjben , please let me know if there is anything else.

**Comment by benjben** [comment]
I wonder if there is a way to automatically get the version of `jackson-core` used to build the fat jar and use this version for `jackson-dataformat-cbor`, in order to remove the manual error-prone step. Maybe @chuwy knows.

Last question: how did you test that the error is gone when you run SSC ?

**Comment by oguzhanunlu** [comment]
> Last question: how did you test that the error is gone when you run SSC ?

@benjben , It is possible to test this without deploying `snowplow-stream-collector-kinesis` to AWS, since exception is thrown when SSC prepares a `DescribeStreamRequest`. What matters here is whether it can prepare a request successfully or not.

First I've reproduced the case by building and running the master branch with a configuration file containing semantically invalid values (values conforming to our pureconfig model but not existing in real life).

Then to test my patch, I built this branch and run using same config file. SSC thrown another exception where the stacktrace showed me that it was able to prepare a request and execute and fail due to invalid security token (I just entered XXXs as accessKey and secretKey).

I can share the config file in case you'd like test it yourself too. In case you don't want to test but wonder the latter exception, here it is.

```
Exception in thread "main" com.amazonaws.services.kinesis.model.AmazonKinesisException: The security token included in the request is invalid. (Service: AmazonKinesis; Status Code: 400; Error Code: UnrecognizedClientException; Request ID: c1e20a21-86ed-cfd8-96f8-3b7ae054188f)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.handleErrorResponse(AmazonHttpClient.java:1712)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeOneRequest(AmazonHttpClient.java:1367)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeHelper(AmazonHttpClient.java:1113)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.doExecute(AmazonHttpClient.java:770)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeWithTimer(AmazonHttpClient.java:744)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.execute(AmazonHttpClient.java:726)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.access$500(AmazonHttpClient.java:686)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutionBuilderImpl.execute(AmazonHttpClient.java:668)
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:532)
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:512)
	at com.amazonaws.services.kinesis.AmazonKinesisClient.doInvoke(AmazonKinesisClient.java:2809)
	at com.amazonaws.services.kinesis.AmazonKinesisClient.invoke(AmazonKinesisClient.java:2776)
	at com.amazonaws.services.kinesis.AmazonKinesisClient.invoke(AmazonKinesisClient.java:2765)
	at com.amazonaws.services.kinesis.AmazonKinesisClient.executeDescribeStream(AmazonKinesisClient.java:875)
	at com.amazonaws.services.kinesis.AmazonKinesisClient.describeStream(AmazonKinesisClient.java:846)
	at com.amazonaws.services.kinesis.AmazonKinesisClient.describeStream(AmazonKinesisClient.java:887)
	at com.snowplowanalytics.snowplow.collectors.scalastream.sinks.KinesisSink$.streamExists(KinesisSink.scala:125)
	at com.snowplowanalytics.snowplow.collectors.scalastream.sinks.KinesisSink$.$anonfun$createAndInitialize$2(KinesisSink.scala:52)
	at scala.util.Either.flatMap(Either.scala:341)
	at com.snowplowanalytics.snowplow.collectors.scalastream.sinks.KinesisSink$.createAndInitialize(KinesisSink.scala:50)
```

File: 2-collectors/scala-stream-collector/project/Dependencies.scala [hunk 1]
```diff
@@ -32,6 +32,10 @@ object Dependencies {
     val slf4j                = "1.7.26"
     val config               = "1.3.4"
     val prometheus           = "0.6.0"
+    /** TODO Make sure cbor version stays aligned with jackson-core
```
Comment (chuwy) [inline 2-collectors/scala-stream-collector/project/Dependencies.scala:35]: 1. It's not a TODO really, it's just a note
2. Could you please put it onto the same line `val cbor = "2.9.10"  //  See snowplow/snowplow/issues/4266

File: 2-collectors/scala-stream-collector/project/Dependencies.scala [hunk 2]
```diff
@@ -32,6 +32,10 @@ object Dependencies {
...
+     *  See snowplow/snowplow/issues/4266
+     */
+    val cbor                 = "2.9.10"
     // Scala
     val collectorPayload     = "0.0.0"
     val scalaz7              = "7.0.9"
```
Comment (chuwy) [inline 2-collectors/scala-stream-collector/project/Dependencies.scala:41]: How do we still have `scalaz7` dependency, @benjben?

File: 2-collectors/scala-stream-collector/project/Dependencies.scala (hunk 1 above)
Comment (oguzhanunlu) [reply 2-collectors/scala-stream-collector/project/Dependencies.scala:35]: Sure, will do now

**Comment by oguzhanunlu** [comment]
Thanks @chuwy , just sent a separate commit for your feedback, please let me know if I can squash it

File: 2-collectors/scala-stream-collector/project/Dependencies.scala (hunk 2 above)
Comment (benjben) [reply 2-collectors/scala-stream-collector/project/Dependencies.scala:38]: It got removed only from SCE, not SSC

Comment (chuwy) [reply 2-collectors/scala-stream-collector/project/Dependencies.scala:38]: We definitely should do it in 1.2.0.

Comment (benjben) [reply 2-collectors/scala-stream-collector/project/Dependencies.scala:38]: Definitely!

**Pull request merged**